ChemCheck/upload/ck2yaml.py

```diff
@@ -1923,10 +1923,14 @@ def convert_mech(input_file, thermo_file=None, transport_file=None,
                      quiet=False, permissive=None):
 
         parser = Parser()
+        f = os.path.split(input_file)[0]
+        p = os.path.join(f, "error.txt")
+        if not p:
```

Comment:
I don't think this line will do what you expect. It looks like you're trying to open the file with write access if it doesn't exist. This only checks if the string `p` is False or empty, not whether or not the file exists. Also, it is generally not a good Python practice to leave open files like this. Why are you opening it but not doing anything with it?